### ISSUE TITLE
Upgrade to Angular 7.3

### ISSUE THREAD

**Issue by williamxsp** [BioPhoton/angular-star-rating#109]
## I'm submitting a...

<pre><code>
[ ] Bug report  
[x] Feature request
</code></pre>

## Current behavior

Im trying to upgrade my system to angular 7.3 but angular-star-rating have some incompatibles peer dependencies (@angular/forms, @angular/core and rjxs)

How can I help you to upgrade these dependencies?

## Minimal reproduction of the problem with instructions

``` ng update angular/core ````

```
Package "angular-star-rating" has an incompatible peer dependency to "@angular/forms" (requires "^4.0.0 || ^5.0.0" (extended), would install "7.2.8").
                  Package "angular-star-rating" has an incompatible peer dependency to "@angular/core" (requires "^4.0.0 || ^5.0.0" (extended), would install "7.2.8").
                  Package "angular-star-rating" has an incompatible peer dependency to "rxjs" (requires "^5.1.0", would install "6.4.0").
```

**Comment by mac1253** [comment]
Would also like to see this upgraded to 7

**Comment by HanumanthDB** [comment]
I am also facing the same error..

**Comment by almaz1213** [comment]
cannot find content of '~css-star-rating/scss/star-rating'

**Comment by theHimanshuShekhar** [comment]
Please upgrade to Angular v8.0.0 now

**Comment by webcat12345** [comment]
after upgrading to 8, i get this

```
RROR in ./src/styles.scss (./node_modules/@angular-devkit/build-angular/src/angular-cli-files/plugins/raw-css-loader.js!./node_modules/postcss-loader/src??embedded!./node_modules/sass-loader/lib/loader.js??ref--13-3!./src/styles.scss)
Module build failed (from ./node_modules/sass-loader/lib/loader.js):

      @extend .star.direction-rtl;
             ^
      Compound selectors may no longer be extended.
Consider `@extend .star, .direction-rtl` instead.
See http://bit.ly/ExtendCompound for details.

    ╷
297 │       @extend .star.direction-rtl;
    │               ^^^^^^^^^^^^^^^^^^^
    ╵
  node_modules/css-star-rating/scss/_elements.scss 297:15  root stylesheet
      in /home/rango/Desktop/ruben/hirescapes-front-end/node_modules/css-star-rating/scss/_elements.scss (line 297, column 15)
```

**Comment by harsh-pamnani** [comment]
@webcat12345 I am also facing the same issue. Did you find any solution for this?

**Comment by webcat12345** [comment]
Just import this `@import "~css-star-rating/css/star-rating.min.css";` not scss mixins. Also you need to override classes to make your own style. Functions won't work.

Something like this.

```
.rating.color-negative .star-container .star {
  svg {
    fill: #f18f01 !important;
  }
  i {
    color: #f18f01 !important;;
  }
}

.rating.color-ok .star-container .star {
  svg {
    fill: #f18f01 !important;;
  }
  i {
    color: #f18f01 !important;;
  }
}

.rating.color-positive .star-container .star {
  svg {
    fill: #f18f01 !important;;
  }
  i {
    color: #f18f01 !important;;
  }
}

.rating.color-default .star-container .star {
  svg {
    fill: #f18f01 !important;;
  }
  i {
    color: #f18f01 !important;;
  }
}
```

**Comment by harsh-pamnani** [comment]
Ok. Thank you so much. I will try with the solution you provided.

**Comment by ProtocolNebula** [comment]
Same error (upgraded from angular 6 to 8).

I tried to fix it but **scss** files need a "full rewrite", I hope the maintainer would fix it (I'm using .scss vars as supposed to work).
In fact, if you use angular 8+ but angular-cli 6, it compiles successfully, which is a bit strange.

After try fixs, new scss versions generate an almost infinite loop or a 450KB file in the best cases.


**EDIT:** To generate the file even with a big size execute (**nodejs 13+**): 
`./node_modules/sass/sass.js path/to/your/star-rating.scss > star-rating.min.css `

And after includes the file (not tested).

**Comment by barryam3** [comment]
This is blocked by BioPhoton/css-star-rating#44 which I have submitted a fix for (BioPhoton/css-star-rating#47) so here's hoping @BioPhoton accepts it. He doesn't seem to have been very active as of late.

**Comment by billyjov** [comment]
Done!
